app/src/main/java/com/stxnext/intranet2/activity/TimeReportActivity.java

```diff
@@ -35,17 +52,61 @@ protected void onCreate(Bundle savedInstanceState) {
         toolbar = (Toolbar) findViewById(R.id.toolbar);
         setSupportActionBar(toolbar);
         getSupportActionBar().setDisplayHomeAsUpEnabled(true);
-        viewContainer = (FrameLayout) findViewById(R.id.container);
-        timeReportTitle = (TextView) findViewById(R.id.time_report_title);
+        timeReports = (LinearLayout) findViewById(R.id.time_reports);
         Calendar month = Calendar.getInstance();
-        timeReportTitle.setText(getString(R.string.time_report_for) + " " + DateFormat.format("MM.yyyy", month));
+        progressView = findViewById(R.id.progress_container);
+        progressView.setVisibility(View.VISIBLE);
+        if (getIntent() != null) {
+            userId = getIntent().getStringExtra(USER_ID_TAG);
+            UserApi userApi = new UserApiImpl(this, this);
+            userApi.getTimeReport(userId, month);
+            month = Calendar.getInstance();
+            month.add(Calendar.MONTH, -1);
+            userApi.getTimeReport(userId, month);
+            month = Calendar.getInstance();
+            month.add(Calendar.MONTH, -2);
+            userApi.getTimeReport(userId, month);
+        }
+    }
+
+    private TableLayout createTimeTable(List<TimeReportDay> timeReportDays) {
```

Comment:
methods that create calendar layout should be refactored into a single class StxCalendar